crates/core/tedge_agent/src/operation_logs.rs

```diff
@@ -38,31 +50,32 @@ impl OperationLogs {
         Ok(operation_logs)
     }
 
-    pub async fn new_log_file(&self, kind: LogKind) -> Result<LogFile, std::io::Error> {
+    pub async fn new_log_file(&self, kind: LogKind) -> Result<LogFile, OperationLogsError> {
         if let Err(err) = self.remove_outdated_logs() {
             // In no case a log-cleaning error should prevent the agent to run.
             // Hence the error is logged but not returned.
             log::warn!("Fail to remove the out-dated log files: {}", err);
         }
 
-        let now = Utc::now();
+        let now = OffsetDateTime::now_utc();
+        let format =
+            format_description::parse("[year]-[month]-[day]T[hour repr:24]:[minute]:[second]z")?;
+
```

Comment:
What's the difference with `format_description::well_known::Rfc3339`?

Using a well known format would avoid to add the new errors kind `FromTimeInvalidFormatDescription(#[from] time::error::InvalidFormatDescription)`

Reply:
These file names are only used internally: we don't have to comply to some external requirements.

Hence, I would prefer to change the names rather than using a non well known format.
* There is no risk of a bug in the agent crate itself. The file names are never parsed. One just need to use a date format which order is compatible with the alphabetic order. And this is the case for Rfc3339 with z or Z.